src/main/java/com/kentington/thaumichorizons/common/items/ItemFocusLiquefaction.java

```diff
@@ -89,105 +89,135 @@ public ItemStack onFocusRightClick(final ItemStack itemstack, final World world,
+                    ItemFocusLiquefaction.soundDelay.put(playerProfile, System.currentTimeMillis() + 1200L);
                 }
             } else {
-                ItemFocusLiquefaction.soundDelay.put(pp, 0L);
+                ItemFocusLiquefaction.soundDelay.put(playerProfile, 0L);
             }
-            if (p.worldObj.isRemote) {
+            if (player.worldObj.isRemote) {
                 ItemFocusLiquefaction.beam.put(
-                        pp,
+                        playerProfile,
                         Thaumcraft.proxy.beamCont(
-                                p.worldObj,
-                                p,
+                                player.worldObj,
+                                player,
                                 tx,
                                 ty,
                                 tz,
                                 0,
                                 16729156,
                                 false,
                                 (impact > 0) ? ((float) size) : 0.0f,
-                                ItemFocusLiquefaction.beam.get(pp),
+                                ItemFocusLiquefaction.beam.get(playerProfile),
                                 5));
             }
-            if ((ent == null || (!(ent instanceof EntityLiving) && (!(ent instanceof EntityItem)
-                    || FurnaceRecipes.smelting().getSmeltingResult(((EntityItem) ent).getEntityItem()) == null)))
-                    && mop != null
-                    && mop.typeOfHit == MovingObjectPosition.MovingObjectType.BLOCK) {
-                final Block bi = p.worldObj.getBlock(mop.blockX, mop.blockY, mop.blockZ);
-                final int md = p.worldObj.getBlockMetadata(mop.blockX, mop.blockY, mop.blockZ);
-                final int meltable = this.isMeltableBlock(bi, md);
-                final boolean flammable = bi
-                        .isFlammable(p.worldObj, mop.blockX, mop.blockY, mop.blockZ, ForgeDirection.UNKNOWN);
-                if (meltable > 0 || flammable
-                        || FurnaceRecipes.smelting().getSmeltingResult(new ItemStack(bi, 1, md)) != null) {
+
+            if (ent instanceof EntityLiving) {
+                if (wand.consumeAllVis(stack, player, ItemFocusLiquefaction.costCritter, true, true)) {
+                    ThaumicHorizons.proxy.smeltFX(ent.posX - 0.5, ent.posY, ent.posZ - 0.5, player.worldObj, 5, false);
+                    ent.attackEntityFrom(DamageSource.inFire, 1.0f + 0.5f * wand.getFocusPotency(stack));
+                }
+            } else if (resultItem != null) {
+                final int quantity = ((EntityItem) ent).getEntityItem().stackSize;
+                if (wand.consumeAllVis(stack, player, this.getVisCost(stack), true, true)) {
+                    ThaumicHorizons.proxy
+                            .smeltFX(ent.posX - 0.5, ent.posY, ent.posZ - 0.5, player.worldObj, 5 * quantity, false);
+
+                    resultItem.stackSize = quantity;
```

Comment:
This sets the output item amount equal to the input item and ignores the recipe output amount. The correct amount would be `inputAmount * recipeOutputAmount`.

Reply:
I agree with you, just don't want to be the one working it.
That's how it previously worked and making it adhere to that sound logic requires a mini rework.